.pre-commit-config.yaml

```diff
@@ -0,0 +1,11 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: check-yaml
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+  - repo: https://github.com/sirwart/ripsecrets
+    rev: v0.1.7
+    hooks:
+      - id: ripsecrets
```

Comment:
This one seems to require a rust installation, and mine wasn't new enough.
```
[INFO] Installing environment for https://github.com/sirwart/ripsecrets.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/home/mark/.cargo/bin/cargo', 'install', '--bins', '--root', '/home/mark/.cache/pre-commit/repohp1cqnt3/rustenv-system', '--path', '.')
return code: 101
stdout: (none)
stderr:
      Installing ripsecrets v0.1.7 (/home/mark/.cache/pre-commit/repohp1cqnt3)
        Updating crates.io index
    error: failed to compile `ripsecrets v0.1.7 (/home/mark/.cache/pre-commit/repohp1cqnt3)`, intermediate artifacts can be found at `/home/mark/.cache/pre-commit/repohp1cqnt3/target`
    
    Caused by:
      package `clap v4.5.16` cannot be built because it requires rustc 1.74 or newer, while the currently active rustc version is 1.70.0
      Either upgrade to rustc 1.74 or newer, or use
      cargo update -p clap@4.5.16 --precise ver
      where `ver` is the latest version of `clap` supporting rustc 1.70.0
Check the log at /home/mark/.cache/pre-commit/pre-commit.log
```

Reply:
I think this is a problem if `rustc` is already installed. It might prioritise the existing installation as opposed to running the one included within `/home/mark/.cache/pre-commit/repohp1cqnt3`

Reply:
If Rust toolchain is detected it will not install Rust for building of pre-commit hooks that have it as a build dependency [1].

Therefore, if an older version is installed it might not be compatible with specific hooks used.

However, ripsecrets v0.1.7 does work with rust 1.7.0 it only fails because Cargo does use the `cargo.lock` by default and requires `--locked` to be passed [2].

This [PR](https://github.com/pre-commit/pre-commit/pull/3163) proposes a simple change to pass `--locked` which I have tested and does work.

Maybe `cargo.conf` could force this [3]?

Not sure how you would like to proceed with this issue.

- [1] https://github.com/pre-commit/pre-commit/blob/main/pre_commit/languages/rust.py#L30
- [2] https://doc.rust-lang.org/cargo/commands/cargo-install.html#:~:text=The%20%2D%2Dlocked%20flag%20can,when%20the%20package%20was%20published.
- [3] https://doc.rust-lang.org/cargo/reference/config.html